### PR TITLE
[Python] Add option to select/detect content-type.

### PR DESCRIPTION
**Pull request by tomplus**
It's follow up the PR #10686 for python-legacy. It adds an extra argument `_content_type` to api calls to force a body content-type. It also extends `select_header_content_type` to return `application/json-patch+json` and help to override it in more complex scenarios.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @Jyhess (2019/01) @arun-nalla (2019/11) @spacether (2019/11)

Thank you.

### CODE REVIEW

File: samples/client/petstore/python-asyncio/.openapi-generator/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-5.3.1-SNAPSHOT
```
Comment (spacether) [inline samples/client/petstore/python-asyncio/.openapi-generator/VERSION:1]: Thank you for your PR. Generally it looks great; can you rebase?
These version file should not be changing.

Comment (tomplus) [reply samples/client/petstore/python-asyncio/.openapi-generator/VERSION]: Sorry about that, fixed.

**Comment by spacether** [comment]
@tomplus you have a CI error for a python test:
FAILED test/test_fake_api.py::TestFakeApi::test_test_endpoint_enums_length_one
Can you fix it?
Once you fix it I can merge it. Your PR looks good otherwise. Thanks!

**Comment by tomplus** [comment]
It looks like all test passed, there is one error reported by ci/bitrise which is probably unrelated to changes.

**Review by spacether — APPROVED** [review]
Thank you for your PR; this looks good!

**Comment by spacether** [comment]
Maven error is unrelated

**Pull request merged**